ludwig/config_validation/checks.py

```diff
@@ -409,3 +417,46 @@ def check_concat_combiner_requirements(config: "ModelConfig") -> None:  # noqa:
             "3D encoder output shapes, or 3) Remove features to ensure that output shapes from all encoders are the "
             "same dimension (all 2D or all 3D)."
         )
+
+
+@register_config_check
+def check_hyperopt_nested_parameter_dicts(config: "ModelConfig") -> None:  # noqa: F821
+    """Checks that all nested parameters in a hyperopt config exist."""
+    if config.hyperopt is None or "." not in config.hyperopt.parameters:
```

Comment:
Still confused by the significance of the single dot

Reply:
We name nested spaces `.` according to the docs. I think we could revisit this design down the road if we expand on nested spaces.

Reply:
Still lost 😓 . Can you give an example?

Reply:
@ksbrar [This](https://ludwig.ai/latest/user_guide/hyperopt/#nested-ludwig-config-parameters) is what Jeff may be referring to.